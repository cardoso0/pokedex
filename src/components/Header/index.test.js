import { render, screen } from "@testing-library/react"
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { BrowserRouter } from 'react-router-dom'

import { Header } from '.'
import { pokedex } from "../../assets/pokedex.png"
import { pokebola } from '../../assets/favicon.png'

const renderHeader = 
  <BrowserRouter>
    <Header />
  </BrowserRouter>

describe("Header Component", () => {
  it("component rendering | snapshot", () => {
    const tree = renderer
      .create(renderHeader)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Header component rendering", () => {
    const tree = renderer
      .create(renderHeader)
    expect(tree).toBeTruthy()
  })

  it('checking logo alt', () => {
    render(renderHeader)
    const logo = screen.getByRole('img')
    expect(logo).toHaveAttribute('alt', 'logo pokedex')
  })

  it('checking logo src', () => {
    render(renderHeader)
    const logo = screen.getByAltText('logo pokedex')
    expect(logo).toHaveAttribute('src', pokedex)
  })

  it('checking pokebola alt', () => {
    render(renderHeader)
    const image = screen.getByAltText('pokebola')
    expect(image).toHaveAttribute('alt', 'pokebola')
  })

  it('checking pokebola src', () => {
    render(renderHeader)
    const image = screen.getByAltText('pokebola')
    expect(image).toHaveAttribute('src', pokebola)
  })
})