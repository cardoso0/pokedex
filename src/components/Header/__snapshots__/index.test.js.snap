// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component component rendering | snapshot 1`] = `
<div
  className="sc-bczRLJ xgqim"
>
  <header
    className="sc-gsnTZi dgMtgR"
  >
    <a
      href="/pokedex/trainers"
      onClick={[Function]}
    >
      <img
        alt="logo pokedex"
        className="logo"
        src={Object {}}
      />
    </a>
    <nav
      className="sc-dkzDqf dwhkCq"
    >
      <a
        href="/pokedex/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/pokedex/favorites"
        onClick={[Function]}
      >
        Menu.fav
      </a>
    </nav>
    <a
      href="/pokedex/favorites"
      onClick={[Function]}
    >
      <img
        alt="pokebola"
        className="pokebola"
        src={Object {}}
      />
    </a>
  </header>
</div>
`;
